ui/src/component/bullet.js

```diff
@@ -0,0 +1,69 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import ReactEcharts from 'echarts-for-react';
+import _ from 'lodash';
+import { validate, getDataOption } from '../utils';
+
+export default class Bullet extends PureComponent {
+  static propTypes = {
+    source: PropTypes.arrayOf(PropTypes.array).isRequired,
+  }
+
+  render() {
+    const { source } = this.props;
+    validate(this.props.source);
+    const dataOption = getDataOption({
+      source,
+      defaultSeriesOpt: (index, total) => {
+        const map = {
+          0: {
+            type: 'bar',
+            barGap: '-300%',
+            barWidth: 6,
+            z: 10,
+            itemStyle: {
+              normal: {
+                color: '#4682b4',
+              },
+            },
+          },
+          1: {
+            type: 'scatter',
+            symbol: 'rect',
+            silent: true,
+            itemStyle: {
+              normal: {
+                color: '#333',
+              },
+            },
+            symbolSize: [20, 3],
+            symbolOffset: ['-30%', 0],
+            z: 20,
+          },
+          default: (idx, tot) => ({
+            type: 'bar',
+            stack: 'total',
+            barWidth: 30,
+            color: (rgb => `rgb(${rgb}, ${rgb}, ${rgb})`)(_.round((idx - 2) / (tot - 2) * (255 - 153) + 153)),
```

Comment:
If you define color in this  way, we will have no way to do theme for this chart.

Reply:
This is not to hard code color here, but use a formula to calculate the gradient color.